### DISCUSSION TITLE
automatically calculate the width of the console

### DISCUSSION THREAD

**Discussion by pheninux** [charmbracelet/bubbletea#276] [Q&A] [answered]
hello ,

how to automatically calculate the width of the console. I want to display data in the center and when I change the width of the console the data remains in the center? 

can you help me please

**Comment by bashbunni** [comment, marked as answer]
Hey @pheninux, 

That would just require that you handle `tea.WindowSizeMsg` in your model's `Update` function. 
```go
	case tea.WindowSizeMsg:
		m.width = msg.Width
		return m, nil
```
You would set your new width in the block of code and after `Update`, `View` will be called to redraw your model. It's in `View` that it will redraw your model based on the updated terminal width. I hope that helps!

**Reply by pheninux** [reply]
Think you very much I will try that

**Reply by bashbunni** [reply]
Let us know how it goes!

**Reply by pheninux** [reply]
hello , 
I tried and it works ,
tks ;)

**Reply by pheninux** [reply]
hi

I have another question, why when I change the size of my console does it redraw the data several times in the console?
<img width="577" alt="Capture d’écran 2022-04-03 à 13 44 22" src="https://user-images.githubusercontent.com/58260568/161426567-3d8b59f3-94e4-454d-9e29-dc228dc1fa34.png">

see screensho
<img width="577" alt="Capture d’écran 2022-04-03 à 13 44 22" src="https://user-images.githubusercontent.com/58260568/161426567-3d8b59f3-94e4-454d-9e29-dc228dc1fa34.png">

<img width="535" alt="Capture d’écran 2022-04-03 à 13 46 01" src="https://user-images.githubusercontent.com/58260568/161426578-878e705d-2265-4dc4-8732-aadf2f752a96.png">

this is my code : 

package main

import (
	"fmt"
	tea "github.com/charmbracelet/bubbletea"
	"github.com/charmbracelet/lipgloss"
	"strings"
)

var (
	with = 200
	// Dialog.
	BoxStyle = lipgloss.NewStyle().
		Border(lipgloss.RoundedBorder()).
		BorderForeground(lipgloss.Color("80")).
		Padding(0, 0).
		BorderTop(true).
		BorderLeft(true).
		BorderRight(true).
		BorderBottom(true)

	forcolor1 = lipgloss.NewStyle().Foreground(lipgloss.Color("98"))
	forcolor2  = lipgloss.NewStyle().Foreground(lipgloss.Color("30"))
	separator = lipgloss.NewStyle().SetString("❣").Foreground(lipgloss.Color("40")).Padding(0,1,0,1).String()
)

type model struct {
	out string
}

func (m *model) Init() tea.Cmd {
	return nil
}

func (m *model) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
	switch message := msg.(type) {
	case tea.KeyMsg:
		switch message.String() {
		case "q":
			return m, tea.Quit
		}

	case tea.WindowSizeMsg:
		with = message.Width
		return m, nil
	}
	return m, nil
}

func (m *model) View() string {

	sb := strings.Builder{}
	sb.WriteString(lipgloss.JoinHorizontal(lipgloss.Center, forcolor1.SetString("Hello").String()+separator+forcolor2.SetString("World").String()))
	m.out = lipgloss.Place(with, 0, lipgloss.Center, lipgloss.Top, sb.String(), lipgloss.WithWhitespaceChars(""))

	return BoxStyle.Width(with - 2).Render(m.out)
}

func main() {

	m := model{}

	if err := tea.NewProgram(&m).Start(); err != nil {
		fmt.Println(err)
	}

}
